src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1325 @@
+                // (8,20): error CS9201: Slice patterns may not be used for a value of type 'int'.
+                //         _ = a is {(..)};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForSlicePattern, "..").WithArguments("int").WithLocation(8, 20),
+                // (9,18): error CS9203: Slice patterns may only be used once and directly inside a list pattern.
+                //         _ = a is ..;
+                Diagnostic(ErrorCode.ERR_MisplacedSlicePattern, "..").WithLocation(9, 18),
+                // (10,19): error CS9203: Slice patterns may only be used once and directly inside a list pattern.
+                //         _ = a is [..];
+                Diagnostic(ErrorCode.ERR_MisplacedSlicePattern, "..").WithLocation(10, 19),
+                // (10,19): error CS9201: Slice patterns may not be used for a value of type 'int'.
+                //         _ = a is [..];
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForSlicePattern, "..").WithArguments("int").WithLocation(10, 19),
+                // (11,24): error CS9203: Slice patterns may only be used once and directly inside a list pattern.
+                //         _ = a switch { .. => 0, _ => 0 };
+                Diagnostic(ErrorCode.ERR_MisplacedSlicePattern, "..").WithLocation(11, 24),
+                // (12,27): error CS9203: Slice patterns may only be used once and directly inside a list pattern.
+                //         switch (a) { case ..: break; }
+                Diagnostic(ErrorCode.ERR_MisplacedSlicePattern, "..").WithLocation(12, 27));
+        }
+
+        [Fact]
+        public void ListPattern_MemberLookup_StaticIndexer()
+        {
+            var vbSource = @"
+Namespace System
+    Public Structure Index
+    End Structure
+End Namespace
+Public Class Test1
+    Public Shared ReadOnly Property Item(i As System.Index) As Integer
+        Get
+            Return 0
+        End Get
+    End Property
+    Public Property Length As Integer = 0
+End Class
+";
+            var csSource = @"
+class X
+{
+    public static void Main()
+    {
+        _ = new Test1() is {0};
+    } 
+}
+";
+            var vbCompilation = CreateVisualBasicCompilation(vbSource);
+            var csCompilation = CreateCompilation(csSource, parseOptions: TestOptions.RegularWithListPatterns, references: new[] { vbCompilation.EmitToImageReference() });
+            // PROTOTYPE(list-patterns) Unsupported because the lookup fails not that the indexer is static
+            csCompilation.VerifyEmitDiagnostics(
+                // (6,28): error CS9200: List patterns may not be used for a value of type 'Test1'.
+                //         _ = new Test1() is {0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForListPattern, "{0}").WithArguments("Test1").WithLocation(6, 28));
+        }
+
+        [Theory]
+        [InlineData("public int this[Index i] { set {} }")]
```

Comment:
Let's add the scenario with private getter (see comment in `PerformPatternIndexerLookup`)

Reply:
>Consider adding indexing access to show matching behavior: _ = new Test1()[^1];

That would make it hard to use inline data :( because indexer access gives different errors.